proposals/NNNN-SDL-device-listener.md

```diff
@@ -0,0 +1,244 @@
+
+The SDL library already attempts to prevent the user from seeing the notification. This is done by quickly enters the foreground as required by the Android OS, then within 10 milliseconds, the service exits the foreground, notification removed, and notification channel deleted. However, based on testing different phones it was obvious that this has a multitude of different results. Some phones do exactly as expected and the notification is cleared before presented to the user. Other devices and OS versions will leave the notification in the tray for roughly 5 seconds until it is removed. The project maintainers have refactored the logic to avoid this for the majority of device and OS combinations, but some devices do not honor the APIs correctly. 
+
+Because of these adverse situations, the library should alter its logic that handles the starting of the `SdlRouterService`. This can be accomplished by creating a new gatekeeper class that prevents the service from ever being started until there has been a confirmed SDL connection via the standard SDL UUID. This logic can be inserted into the `SdlBroadcastReceiver` without a lot of overhead to existing code.
+
+## Proposed solution
+
+The solution will be to create a temporary bluetooth server socket in the broadcast receiver that receives the bluetooth connection intents. A new thread will be spawned off that will handle the listening. If a connection is made with the device, the bluetooth connection will be closed, and then the `SdlRouterService` will be started up directly to the foreground. A timeout handler will be present to keep the socket alive for a short duration without allowing it to run forever. Once the timeout expires, if there was no connection made, the socket will stop listening and the router service will not be started. In either case the MAC address will be saved into preferences with the results of a successful connection or not. 
+
+On the next connection with that same device, via the MAC address, we can avoid this step if the device has been recognized as an SDL enabled device. If the device has not made a connection, the timeout window can be shortened. 
+
+Leaving a small window of time to allow for connections at this stage is very important. The first time a device is connected, it might be plausible the SDL connection was just too slow and the following connection might catch the right timeout. The goal is make sure there isn't a missed SDL enabled device while taking into consideration that most users do not interact with such devices.
```

Comment:
```suggestion
Leaving a small window of time to allow for connections at this stage is very important. The first time a device is connected, it might be plausible the SDL connection was just too slow and the following connection might catch the right timeout. The goal is to make sure there isn't a missed SDL enabled device while taking into consideration that most users do not interact with such devices.
```